### PR TITLE
[ACR] Add public network access

### PR DESCRIPTION
**Pull request by djyou**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by yugangw-msft — APPROVED** [review]

**Comment by yugangw-msft** [comment]
This is an update to a preview API

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json [hunk 1]
```diff
@@ -2130,6 +2143,18 @@
...
           "description": "Enable a single data endpoint per region for serving data.",
           "type": "boolean"
+        },
+        "publicNetworkAccess": {
+          "description": "Whether or not public network access is allowed for the container registry.",
+          "enum": [
+            "Enabled",
+            "Disabled"
+          ],
+          "type": "string",
```
Comment (fengzhou-msft) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json:2153]: Why choose enum over a boolean property like `publicNetworkAccessEnabled` as I see there's `adminUserEnabled` and `dataEndpointEnabled` in existing properties?

Comment (djyou) [reply specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-12-01-preview/containerregistry.json:2153]: Just being consistent with all other RPs for the same feature https://github.com/Azure/azure-rest-api-specs/search?q=publicNetworkAccess

**Comment by yugangw-msft** [comment]
@fengzhou-msft, could you please take a look, approve and merge it if you don't have other concern?

**Review by pilor — APPROVED** [review]
:shipit:

**Comment by fengzhou-msft** [comment]
@akning-ms @NullMDR can you help force merge this PR? The breaking change is for an update in a preview API.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**